File: app/src/main/java/io/github/zwieback/familyfinance/business/operation/activity/exception/IllegalOperationTypeException.java
package io.github.zwieback.familyfinance.business.operation.activity.exception;

import io.github.zwieback.familyfinance.core.model.OperationView;

public class IllegalOperationTypeException extends IllegalArgumentException {

    private static final long serialVersionUID = 4171955715539750284L;

    public static IllegalOperationTypeException unsupportedOperationType(OperationView operation) {
        return new IllegalOperationTypeException("Operation with id " + operation.getId() +
                " has unsupported type: " + operation.getType());
    }

    public static IllegalOperationTypeException notTransferOperation(OperationView operation) {
        return new IllegalOperationTypeException("Transfer operation with id " + operation.getId() +
                " must be transfer operation, not " + operation.getType());
    }

    private IllegalOperationTypeException(String message) {
        super(message);
    }
}

File: app/src/main/java/io/github/zwieback/familyfinance/business/operation/activity/exception/IllegalOperationTypeException.kt
package io.github.zwieback.familyfinance.business.operation.activity.exception

import io.github.zwieback.familyfinance.core.model.OperationView

class IllegalOperationTypeException private constructor(message: String) :
    IllegalArgumentException(message) {

    companion object {
        private const val serialVersionUID = 4171955715539750284L

        fun notTransferOperation(operation: OperationView): IllegalOperationTypeException {
            return IllegalOperationTypeException(
                "Transfer operation with id ${operation.id} " +
                        "must be transfer operation, not ${operation.type}"
            )
        }
    }
}
